agent/engine/task_manager.go

```diff
@@ -1040,19 +1066,7 @@ func (mtask *managedTask) progressTask() {
 	// complete, but keep reading events as we do. in fact, we have to for
 	// transitions to complete
 	mtask.waitForTransition(transitions, transitionChange, transitionChangeEntity)
-	// update the task status
-	if mtask.UpdateStatus() {
-		logger.Info("Container or resource change also resulted in task change", logger.Fields{
-			field.TaskID: mtask.GetID(),
-		})
-
-		// If knownStatus changed, let it be known
-		var taskStateChangeReason string
-		if mtask.GetKnownStatus().Terminal() {
-			taskStateChangeReason = mtask.Task.GetTerminalReason()
-		}
-		mtask.emitTaskEvent(mtask.Task, taskStateChangeReason)
-	}
+	mtask.manageTaskStatusChange()
```

Comment:
Hmm should we move this up? At this point we've already started container transitions (line 1022), but we want to invoke AppNet draining before any of that happen.

Reply:
I will move this to `stopContainer` since it will be easier to UT.